src/multiPortConnector.js

```diff
@@ -0,0 +1,254 @@
+                }
+            });
+        }
+    };
+
+    fluid.defaults("youme.multiPortConnector.inputs", {
+        gradeNames: ["youme.multiPortConnector", "youme.messageReceiver"],
+
+        dynamicComponents: {
+            connection: {
+                type: "youme.connection.input",
+                options: {
+                    listeners: {
+                        "onActiveSense.relay": "{youme.multiPortConnector.inputs}.events.onActiveSense.fire",
+                        "onAftertouch.relay": "{youme.multiPortConnector.inputs}.events.onAftertouch.fire",
+                        "onClock.relay": "{youme.multiPortConnector.inputs}.events.onClock.fire",
+                        "onContinue.relay": "{youme.multiPortConnector.inputs}.events.onContinue.fire",
+                        "onControl.relay": "{youme.multiPortConnector.inputs}.events.onControl.fire",
+                        "onMessage.relay": "{youme.multiPortConnector.inputs}.events.onMessage.fire",
+                        "onNoteOff.relay": "{youme.multiPortConnector.inputs}.events.onNoteOff.fire",
+                        "onNoteOn.relay": "{youme.multiPortConnector.inputs}.events.onNoteOn.fire",
+                        "onPitchbend.relay": "{youme.multiPortConnector.inputs}.events.onPitchbend.fire",
+                        "onProgram.relay": "{youme.multiPortConnector.inputs}.events.onProgram.fire",
+                        "onRaw.relay": "{youme.multiPortConnector.inputs}.events.onRaw.fire",
+                        "onReset.relay": "{youme.multiPortConnector.inputs}.events.onReset.fire",
+                        "onSongPointer.relay": "{youme.multiPortConnector.inputs}.events.onSongPointer.fire",
+                        "onSongSelect.relay": "{youme.multiPortConnector.inputs}.events.onSongSelect.fire",
+                        "onStart.relay": "{youme.multiPortConnector.inputs}.events.onStart.fire",
+                        "onStop.relay": "{youme.multiPortConnector.inputs}.events.onStop.fire",
+                        "onSysex.relay": "{youme.multiPortConnector.inputs}.events.onSysex.fire",
+                        "onTuneRequest.relay": "{youme.multiPortConnector.inputs}.events.onTuneRequest.fire"
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.defaults("youme.multiPortConnector.outputs", {
+        gradeNames: ["youme.multiPortConnector", "youme.messageSender"],
+
+        direction: "outputs",
+
+
+        invokers: {
+            relayEvent: {
+                funcName: "youme.multiPortConnector.outputs.relayEvent",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"] // eventName, payload
+            }
+        },
+
+        dynamicComponents: {
+            connection: {
+                type: "youme.connection.output",
+                options: {
+                    // If use this, we only ever hear from one input, where the relay approach allows it.
+                    // listeners: {
```

Comment:
This works for inputs, so presumably for outputs too?

Reply:
We should pair on it, if I use the sensible commented-out relay strategy, all of the outputs only receive messages from one input.

Reply:
I added tests and confirmed that it works in isolation (only inputs, only outputs) as well as in combination (inputs connected to outputs).  You'll see in the tests that I had to resort to `setTimeout` for one of the tests, because I can't wait for all dynamic children to fire a given event.  This would also have been a problem in the IoC world as far as I know.